packages/dev/core/src/Audio/v2/IAudioNodeParent.ts

```diff
@@ -0,0 +1,10 @@
+/* eslint-disable babylonjs/available */
+/* eslint-disable jsdoc/require-jsdoc */
+
+import type { AbstractAudioNode } from "./abstractAudioNode";
+import type { IDisposable } from "../../scene";
+
+export interface IAudioNodeParent extends IDisposable {
+    _addChildNode(node: AbstractAudioNode): void;
```

Comment:
Getting further into this PR, I think I understand the intent. This is only "public" so that classes that implement this interface (specifically `AbstractAudioNode`) can call `_addChildNode` when their `parent` is set, correct? It's not actually intended to be public in the API, yes? If that is the case, then you could use an abstract base class that exposes protected static functions for adding/removing child nodes. This way it keeps these out of the public contract, but still allows derived classes to mutate the child nodes. I created a quick TS Playground [example here](https://www.typescriptlang.org/play/?target=7#code/PTCWFsAcHsCcBcAEBvACrakDOB1U8ALAYQNABsATWAUwDsAaRI6KaWu+AFQE9JrGAYgFdaAY3ig2zVu1rxGQrNSIBDMmQBGK0QGtG4auGgBfRADMM4RACIa2+NYDcAKBAQYCFKjIr4ZuOCMAMrw3GTU6JiMAGqg1ADuIWH8iAAyKtzQQvAkKrQA5tQAogBuHDFx8Yzo1FhYKhrhFQmR2NU0dQ3hIb7KapraOjx8phYsNnbiALS0vqBlTs7OomxY0OEAdKLhKrAAFACULs4NWPCw9ojbKnWIAIIaZxfid0IUkgBy0BQRuxwozkQQMQkFg816iAA+qJSJQvj8sAB+ABciCC1HgAB4Hk97K93tB4dQAHzHYEgjDwajiagUKEqCgUEjkChEva0b7UVE4854t6fTkHAHk8mgMyIPYAQkIoCwG2hsNZnKwQuQgJFGplcoVLKJWEQAF5EOx4miMYcXBqgcYllbEFr5TDdcqNgyKOzBZbgTb1UDQdAqTS6ZCaEYysy4ZyPT9uY9eS9+YTBaiStBQHS1XaHTrIwjERsfuEqdHqEdfYgfeT-YGqXSznNRPTGRGlT89pA-nJY7iEwSiahO-JjZzu-H4PiBT9VeWq4P5W6W2yOVOvdbbcDq9Ta4h6xJGyHDNBw4q2R2aF37nHnuPE-3B4xl1zLz2b33BcKrWeOPLQ0flCeo0fMtyRtH1TjHK4fFuHlrwnJMfkQagAA8qVoCh9Rgvk3x+AdzyQTMNyERpQEbOwKDYMhuEQgpQHYUdYMTIoaPYQ1jSEdRJVXCkAy3WkoS-OQUWfMc4LvPCyUI4jG0KJABPgQ56Kwydfjwj8NRoeAhFgWh7VIbU5MQRFEV02UNjofJaOoLjK0ksgSJ3DEQUHdtB0U3tlNw-4AB9EBEH4zEsihpztMUJWzAyDUipy8OCu1gQ0rTaC4kD1w1TD3PglTvwPMN-2dNsHTkxgtWAq1wsHVi5OS4F0tfDy50hBcAIKvSNiKkyVWs5wwKvS5rmg3qMqYiyWOQ1D0OEhjsKyuQAW6pYi2HGNJqUzLWNodiyE41xgEQL4KXBKkoTMERxEkWh9WQmAlAoZxHw2ZwgA).

Reply:
I like the suggestion of adding `AbstractAudioNodeParent`, but I don't like the static functions, yet. I'm not sure the extra code is worth making the TypeScript API cleaner if the resulting JavaScript API stays the same.